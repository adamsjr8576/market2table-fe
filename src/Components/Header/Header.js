import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Header.scss';
import { addZipCode, addVendors, addSelectedMarket, addMarkets, addSelectedVendor } from '../../actions';


export const Header = ({ history, path }) => {

const { zipCode, selectedMarket, vendors } = useSelector(state => state);
const dispatch = useDispatch();

const handleBackToMarketInfo = () => {
  history.push(`/markets/${selectedMarket}`);
  dispatch(addVendors([]));
}

const handleBacktoMarkets = () => {
  history.push('/markets');
}

let vendorHomeButton;
if (path.includes('account')) {
  vendorHomeButton = (
    <Link to='/' className='link-to-vendors'>
      <button className='vendor-login-button' onClick={() => {dispatch(addVendors([])); dispatch(addSelectedMarket('')); dispatch(addZipCode('')); dispatch(addMarkets([])); dispatch(addSelectedVendor({}))}}>Home</button>
    </Link>)
} else {
  vendorHomeButton = (
    <Link to='/vendor/account' className='link-to-vendors'>
      <button className='vendor-login-button' onClick={() => { dispatch(addVendors([])); dispatch(addZipCode('')); dispatch(addSelectedMarket('')); dispatch(addSelectedVendor({}));dispatch(addMarkets([]))}}>Vendor Login</button>
    </Link>)
}

let zipCodeButton;
if (path.includes('markets')) {
  zipCodeButton = (
    <button className='back-button' onClick={() => dispatch(addZipCode(''))}>Reset Zip Code</button>
  )
}

if (path.includes('markets') && selectedMarket) {
  zipCodeButton = (
  <button className='back-button' id='back-to-markets-button' onClick={() => {dispatch(addSelectedMarket('')); handleBacktoMarkets()}}>Back To Markets</button>
  )
}

if (path.includes('markets') && path.includes('vendors')) {
  zipCodeButton = (
    <button onClick={handleBackToMarketInfo} className='back-button' id='back-to-market-button'>Back to Market</button>
  )
}

let zipCodeButtonContainer;
if (path.includes('markets')) {
  zipCodeButtonContainer = (
    <>
      <p className='zip-code-par'>Zip Code: {zipCode}</p>
      {zipCodeButton}
    </>
  )
}

  return (
    <header className='main-header'>
      <section className='zip-code-section'>
        {zipCodeButtonContainer}
      </section>
      <h1 className='header-title'>Market 2 Table</h1>
      {vendorHomeButton}
    </header>
  )
}

export default withRouter(Header)

Header.propTypes = {
  history: PropTypes.object,
  zipCode: PropTypes.string,
  selectedMarket: PropTypes.string
}
