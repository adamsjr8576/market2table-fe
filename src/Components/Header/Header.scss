@import './src/variables';

.main-header {
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
  height: 15vh;
}

.header-title {
  color: cornsilk;
  margin: 0;
  font-size: 4em;
  text-align: center;
  width: 70%;
  text-decoration: underline;
}

.zip-code-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 175px;
  height: 85%;
}

.zip-code-par {
  margin: 0;
  margin-top: 20px;
  padding: 5px;
  font-size: 1.2em;
  font-weight: 600;
  letter-spacing: 0.5px;
  color: cornsilk;
  border-radius: 10px;
  width: 90%;
  min-width: 125px;
  text-align: center;
}

.back-button {
  width: 80%;
  color: cornsilk;
  font-weight: 500;
  border-bottom-color: cornsilk;
  background-color: transparent;
  letter-spacing: 0.5px;
  min-width: 145px;
  padding: 5px;
  font-size: 1em;
  // border-radius: 20px;
  // border: 0;
  border-top: none;
  border-right: none;
  border-left: none;
  outline: 0;
  cursor: pointer;
}

.back-button:hover,
.back-button:focus {
  // background-color: $list_hover_color;
  border: 2px solid khaki;
  border-radius: 20px;
  // font-weight: 00;
  // color: black;
  // border: none;
  transform: scale(1.05);
}

.vendor-login-button {
  background-color: transparent;
  color: cornsilk;
  padding: 5px;
  border-radius: 20px;
  font-weight: 500;
  width: 100%;
  font-size: 1.2em;
  letter-spacing: 1px;
  border: 0;
  outline: 0;
  cursor: pointer;
}

.vendor-login-button:hover,
.vendor-login-button:focus {
  border: 2px solid khaki;
  transform: scale(1.05);
}

.vendor-login-button:active {
  transform: scale(.99);
}

.link-to-vendors {
  width: 10%;
  min-width: 125px;
  margin-right: 35px;
  outline: 0;
  text-decoration: none;
}

.vendor-back-button{
  border-radius: 10px;
}

@media screen and (max-width: 750px) {
  .main-header {
    display: grid;
    grid-template-areas:
      "title title"
      "buttons vendorLogin";
    grid-template-rows: auto;
    grid-template-columns: 65% 35%;
    height: 19vh;
    justify-items: center;
  }

  .zip-code-section {
    flex-direction: row;
    width: 100%;
    grid-area: buttons;
  }

  .header-title {
    width: 100%;
    font-size: 3em;
    grid-area: title;
  }

  .vendor-login-button {
    grid-area: vendorLogin;
  }

  .zip-code-par {
    width: 35%;

  }

  .back-button {
    width: 35%;
    
  }
}
