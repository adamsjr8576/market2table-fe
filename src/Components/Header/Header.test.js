import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
  zipCode: '80401',
  selectedMarket: '1923782',
  vendors: [{name: 'Friendly Famrs', id:'1231', image_link: 'image.png'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('Header', () => {

  let wrapper;
  let mockAddZipCode = jest.fn();
  let mockAddVendors = jest.fn();
  let mockAddSelectedMarket = jest.fn();
  let mockAddMarkets = jest.fn();
  let mockAddSelectedVendor = jest.fn();
  let mockHistory = {push: jest.fn()};
  let mockPath;

  it('should match the snapshot with a path that includes markets', () => {
    mockPath = '/markets/234124'
    wrapper = shallow(<Header
      addZipCode={mockAddZipCode}
      addVendors={mockAddVendors}
      addSelectedMarket={mockAddSelectedMarket}
      addMarkets={mockAddMarkets}
      addSelectedVendor={mockAddSelectedVendor}
      history={mockHistory}
      path={mockPath}
       />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke history.push on back to markets button', () => {
    mockPath = '/markets/234124'
    wrapper = shallow(<Header
      addZipCode={mockAddZipCode}
      addVendors={mockAddVendors}
      addSelectedMarket={mockAddSelectedMarket}
      addMarkets={mockAddMarkets}
      addSelectedVendor={mockAddSelectedVendor}
      history={mockHistory}
      path={mockPath}
       />);
    wrapper.find('#back-to-markets-button').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith('/markets');
  });

  it('should match the snapshot with a path that includes account', () => {
    mockPath = '/account/234124'
    wrapper = shallow(<Header
      addZipCode={mockAddZipCode}
      addVendors={mockAddVendors}
      addSelectedMarket={mockAddSelectedMarket}
      addMarkets={mockAddMarkets}
      addSelectedVendor={mockAddSelectedVendor}
      history={mockHistory}
      path={mockPath}
       />);

    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with a path that includes markets and vendors', () => {
    mockPath = '/markets/234124/vendors'
    wrapper = shallow(<Header
      addZipCode={mockAddZipCode}
      addVendors={mockAddVendors}
      addSelectedMarket={mockAddSelectedMarket}
      addMarkets={mockAddMarkets}
      addSelectedVendor={mockAddSelectedVendor}
      history={mockHistory}
      path={mockPath}
       />);

    expect(wrapper).toMatchSnapshot()
  });

  it('should invoke history.path on button click', () => {
    mockPath = '/markets/234124/vendors'
    wrapper = shallow(<Header
      addZipCode={mockAddZipCode}
      addVendors={mockAddVendors}
      addSelectedMarket={mockAddSelectedMarket}
      addMarkets={mockAddMarkets}
      addSelectedVendor={mockAddSelectedVendor}
      history={mockHistory}
      path={mockPath}
       />);

    wrapper.find('#back-to-market-button').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith('/markets/1923782');
  });
});
