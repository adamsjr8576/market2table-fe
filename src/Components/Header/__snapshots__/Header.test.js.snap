// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with a path that includes account 1`] = `
<header
  className="main-header"
>
  <section
    className="zip-code-section"
  />
  <h1
    className="header-title"
  >
    Market 2 Table
  </h1>
  <Link
    className="link-to-vendors"
    to="/"
  >
    <button
      className="vendor-login-button"
      onClick={[Function]}
    >
      Home
    </button>
  </Link>
</header>
`;

exports[`Header should match the snapshot with a path that includes markets 1`] = `
<header
  className="main-header"
>
  <section
    className="zip-code-section"
  >
    <p
      className="zip-code-par"
    >
      Zip Code: 
      80401
    </p>
    <button
      className="back-button"
      id="back-to-markets-button"
      onClick={[Function]}
    >
      Back To Markets
    </button>
  </section>
  <h1
    className="header-title"
  >
    Market 2 Table
  </h1>
  <Link
    className="link-to-vendors"
    to="/vendor/account"
  >
    <button
      className="vendor-login-button"
      onClick={[Function]}
    >
      Vendor Login
    </button>
  </Link>
</header>
`;

exports[`Header should match the snapshot with a path that includes markets and vendors 1`] = `
<header
  className="main-header"
>
  <section
    className="zip-code-section"
  >
    <p
      className="zip-code-par"
    >
      Zip Code: 
      80401
    </p>
    <button
      className="back-button"
      id="back-to-market-button"
      onClick={[Function]}
    >
      Back to Market
    </button>
  </section>
  <h1
    className="header-title"
  >
    Market 2 Table
  </h1>
  <Link
    className="link-to-vendors"
    to="/vendor/account"
  >
    <button
      className="vendor-login-button"
      onClick={[Function]}
    >
      Vendor Login
    </button>
  </Link>
</header>
`;
